### PR TITLE
Busy-wait with callback

### PR DESCRIPTION
**Pull request by magnumripper**
Currently only implemented in nt-opencl, will try tezos next.

### CODE REVIEW

**Comment by solardiz** [comment]
If this is successful, are we going to drop our previous usage of `SIGUSR2` and reuse it like this? Or use another signal? Or move away from signals for this (to other `select` events)?

**Comment by solardiz** [comment]
Also, was `clSetEventCallback` introduced long enough ago and universally enough (across OpenCL implementations) that we can use it unconditionally (a compile-time dependency at least)?

**Comment by magnumripper** [comment]
It's OpenCL 1.2 so should be fine. I'm not sure about repurposing USR2 or find some other signal, first let's see if this is usable at all.

File: src/opencl_common.h [hunk 1]
```diff
@@ -337,6 +337,22 @@ void opencl_process_event(void);
...
+	if (opencl_avoid_busy_wait[gpu_id] && !sleep_event) { \
+		opencl_sleep_pid = getpid(); \
+		sleep_event = &opencl_sleep_event; \
+	}
+
+#define WAIT2_SLEEP(sleep_event)	  \
+	if (sleep_event == &opencl_sleep_event) { \
+		BENCH_CLERROR(clSetEventCallback(opencl_sleep_event, CL_COMPLETE, &opencl_event_callback, NULL), "set callback"); \
+		opencl_sleep(); \
+	}
```
Comment (magnumripper) [inline src/opencl_common.h]: Note to self: Need to drop BENCH_CLERROR here, and instead only sleep if `clSetEventCallback` succeeds.

**Comment by magnumripper** [comment]
I'm giving up on this for now.

**Pull request closed without merging**